dbms/src/DataTypes/DataTypesDecimal.cpp

```diff
@@ -23,42 +25,46 @@ namespace ErrorCodes
     extern const int ARGUMENT_OUT_OF_BOUND;
 }
 
-
-bool decimalCheckComparisonOverflow(const Context & context) { return context.getSettingsRef().decimal_check_overflow; }
-bool decimalCheckArithmeticOverflow(const Context & context) { return context.getSettingsRef().decimal_check_overflow; }
-
-
 //
 
 template <typename T>
 std::string DataTypeDecimal<T>::doGetName() const
 {
     std::stringstream ss;
-    ss << "Decimal(" << precision << ", " << scale << ")";
+    ss << "Decimal(" << this->precision << ", " << this->scale << ")";
```

Comment:
What was wrong here?

Reply:
well, that fails to compile on wast majority of compilers: https://godbolt.org/z/ns7TwF

Reply:
Doesn't `using BaseClass::scale` in derived class helps?